### ISSUE TITLE
Blood DK should recommend Marrowrend in more situations.

### ISSUE THREAD

**Issue by Alchemooney** [Hekili/hekili#1091]
Please describe your idea or feature and when it would be useful.
Tested in single target against a training dummy. 
Hekili recommends using Marrowrend immediately after Bone Shield (the buff it creates) wears off. This is a stacking buff and thanks to the Ossuary passive ability (baseline), blood DK wants to remain at 5 stacks or higher. Since the blood DK will also want to use Death Strikes in a reactionary manner (to damage taken, deviating from the Hekili recommendations), ideally a blood DK would limit the risk of Bone Shield either falling off or stacks dropping too low.

Consequently, I suggest the following changes:
If Bone Shield has 6 charges or less and Runic Power is at least 20 below max, Marrowrend should be very high priority.
If Bone Shield has 6 seconds or less remaining and Runic Power is at least 20 below max, Marrowrend should be next.
If Bone Shield has 6 seconds or less remaining and Runic Power is less than 20 below max, priority should be Death Strike then Marrowrend.
The cost of being 1-second late refreshing is too high, so these rotations provide a little leeway while getting the buff refreshed in time.
Ideally, another feature to add would be if the projected combat time remaining is less than 3 seconds, Marrowrend should be very high priority because this will refresh the buff for the run to the next pack.

Below is a link to my pastebin log.
https://pastebin.com/k851yahV

**Comment by Hekili** [comment]
You're training against a Training Dummy.  Your mitigation action list is called when you are `tanking` and/or you have taken some damage in the last 5 seconds (`incoming_damage_5s > 0`).

If you're not tanking or taking damage, then yes, Marrowrend doesn't get used to keep Bone Shield up.

You can edit this in your priority by going to /hekili > Priorities > Blood > Action Lists > default and editing the second entry, which is Call Action List, and deleting the conditions (`tanking | incoming_damage_5s > 0`).  Hit Accept after wiping out the Conditions.

Have you tried while tanking?

**Comment by Alchemooney** [comment]
Just tried using a tanking dummy so it would hit back. Turns out the recommendation is much better than expected.
Time for refreshing is efficiently managed. I'd suggest a much smaller adjustment then. 
Wipe out recommendations 2 & 3 regarding the timer.

Just looking at suggestion 1, I'd still make a case for using marrowrend at 6 stacks. If we're actively tanking, then using it six stacks would give us a much better safety margin, especially if we've got runes to spare.

I grabbed another shot showing recommendations for HS while at 6 stacks with incoming damage and full or nearly full runes (meaning the earlier use of Marrowrend would set them to recharging sooner, and thus be more efficient).

Single-target tanking, so HS wouldn't be getting any cleave benefit.

Here's the pastebin:
https://pastebin.com/NQQRcRbX
